src/mcp23017.cpp

```diff
@@ -1,33 +1,96 @@
 #include "mcp23017.h"
-#include <errno.h>
-
 
 // TODO (optional): Define macros for useful register below:
-
+#define IODIRA 0x00
+#define IODIRB 0x01
+#define GPIOA 0x12
+#define GPIOB 0x13
 
 // TODO: Initialize i2cBus member
-Mcp23017::Mcp23017(int addr, I2C* i2cBus) {
-
+Mcp23017::Mcp23017(int addr) {
+    this->addr = addr;
 }
 
 uint8_t Mcp23017::get_dir(int pin) {
-    return 0;
+    Wire.requestFrom(IODIRA, 1);
+    int w = 0;
```

Comment:
Typically on the Firmware team, we want our variable names to be descriptive to help with code readability. Try naming it `byte` or `value` instead.